sao/src/view/tree.js

```diff
@@ -985,6 +984,38 @@
             this.record = record;
             // TODO update_children
         },
+        update_visible: function() {
+            var to_hide = [];
+            var to_show = [];
+            for (var i = 0; i < this.columns.length; i++) {
+                var column = this.columns[i];
+                if (column.visible && column.header.css('display') == 'none') {
+                    to_hide.push(i);
+                } else {
+                    to_show.push(i);
+                }
+            }
+            const make_selector = (col_idx) => {
+                // Take into account the selection or optional column
+                var offset = 1;
+                if (this.draggable) {
+                    offset += 1;
+                } else if (this.optionals.length) {
+                    offset += 1;
+                }
+                // CSS is 1-indexed
+                return `tr td:nth-child(${col_idx + offset + 1})`;
+            };
+
+            if (to_hide.length) {
+                this.tbody.find(to_hide.map(make_selector).join(','))
+                    .addClass('invisible').hide();
+            }
+            if (to_show.length) {
+                this.tbody.find(to_show.map(make_selector).join(','))
+                    .removeClass('invisible').show();
```

Comment:
Petit problème ici avec jQuery qui repasse un coup de recalculate style sur tout le DOM en itérant sur le body j'ai l'impression. J'ai cherché des solutions pour éviter ça mais no luck.

Reply:
Je pensais naïvement qu'en utilisant un selecteur ça ne ferait qu'une seule invalidation.